linera-execution/src/runtime.rs

```diff
@@ -1001,18 +1002,20 @@ impl<UserInstance> BaseRuntime for SyncRuntimeInternal<UserInstance> {
     }
 
     fn read_data_blob(&mut self, hash: &CryptoHash) -> Result<Vec<u8>, ExecutionError> {
-        let blob_id = BlobId::new_data_from_hash(*hash);
+        let blob_id = BlobId::new(*hash, BlobType::Data);
         self.transaction_tracker
             .replay_oracle_response(OracleResponse::Blob(blob_id))?;
         let blob_content = self
             .execution_state_sender
             .send_request(|callback| ExecutionRequest::ReadBlobContent { blob_id, callback })?
             .recv_response()?;
-        Ok(blob_content.bytes)
+        Ok(blob_content
+            .into_inner_data()
+            .expect("Should be a data blob!"))
```

Comment:
We wouldn't need that if we didn't serialize the blob type discriminator in the value, too.